homeassistant/components/plex/button.py

```diff
@@ -0,0 +1,68 @@
+"""Representation of Plex buttons."""
+from __future__ import annotations
+
+import datetime
+
+from homeassistant.components.button import ButtonEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.dispatcher import async_dispatcher_send
+from homeassistant.helpers.entity import DeviceInfo, EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.event import async_track_time_interval
+
+from .const import CONF_SERVER_IDENTIFIER, DOMAIN, PLEX_UPDATE_PLATFORMS_SIGNAL
+
+CLIENT_SCAN_INTERVAL = datetime.timedelta(minutes=10)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up Plex button from config entry."""
+    server_id: str = config_entry.data[CONF_SERVER_IDENTIFIER]
+    async_add_entities([PlexScanClientsButton(server_id)])
+
+
+class PlexScanClientsButton(ButtonEntity):
+    """Representation of a scan_clients button entity."""
+
+    _attr_should_poll = False
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_name = "Scan Clients"
+
+    def __init__(self, server_id: str) -> None:
+        """Initialize a scan_clients Plex button entity."""
+        self.server_id = server_id
+        self._attr_unique_id = f"plex-scan_clients-{self.server_id}"
+
+    async def async_added_to_hass(self) -> None:
+        """Call when entity is added."""
+
+        async def async_scheduled_press(now: datetime.datetime):
```

Comment:
Why should this be done in the entity rather than in the main integration? Nifty way for sure but a bit unorthodox?

Reply:
Yeah, I had second thoughts immediately after implementing it this way. In my mind disabling the entity was a simple way for users to disable the automatic polling if they didn't want it for some reason. However that would (wait for it...) disable the entity and take away the ability to trigger it manually. Probably makes sense to keep them separate and stop trying to be clever. 🙃

Reply:
You could add a switch for the timing functionality

Reply:
Split off the automatic schedule and added tests for both points of entry.